src/Gizmos/planeRotationGizmo.ts

```diff
@@ -51,48 +62,76 @@ export class PlaneRotationGizmo extends Gizmo {
-        rotationMesh.material = coloredMaterial;
+        // Build mesh on root node
+        this._gizmoMesh = new Mesh("", gizmoLayer.utilityLayerScene);
+        const { rotationMesh, collider } = this._createGizmoMesh(this._gizmoMesh, thickness, tessellation);
 
-        // Position arrow pointing in its drag axis
-        rotationMesh.rotation.x = Math.PI / 2;
-        drag.rotation.x = Math.PI / 2;
-        parentMesh.addChild(rotationMesh);
-        parentMesh.addChild(drag);
-        parentMesh.lookAt(this._rootMesh.position.add(planeNormal));
+        // Setup Rotation Circle
+        const rotationCirclePaths: any[] = [];
+        this._rotationCircle = this.setupRotationCircle(rotationCirclePaths, this._gizmoMesh);
 
-        this._rootMesh.addChild(parentMesh);
-        parentMesh.scaling.scaleInPlace(1 / 3);
+        this._gizmoMesh.lookAt(this._rootMesh.position.add(planeNormal));
+        this._rootMesh.addChild(this._gizmoMesh);
+        this._gizmoMesh.scaling.scaleInPlace(1 / 3);
         // Add drag behavior to handle events when the gizmo is dragged
         this.dragBehavior = new PointerDragBehavior({ dragPlaneNormal: planeNormal });
         this.dragBehavior.moveAttached = false;
         this.dragBehavior.maxDragAngle = Math.PI * 9 / 20;
         this.dragBehavior._useAlternatePickedPointAboveMaxDragAngle = true;
         this._rootMesh.addBehavior(this.dragBehavior);
 
-        var lastDragPosition = new Vector3();
+        // Closures for drag logic
+        let dragDistance = 0;
+        const lastDragPosition = new Vector3();
+        let dragPlanePoint = new Vector3();
+        const rotationMatrix = new Matrix();
+        const planeNormalTowardsCamera = new Vector3();
+        let localPlaneNormalTowardsCamera = new Vector3();
 
         this.dragBehavior.onDragStartObservable.add((e) => {
             if (this.attachedNode) {
                 lastDragPosition.copyFrom(e.dragPlanePoint);
+
+                // This is for instantiation location of rotation circle
+                // Rotation Circle Forward Vector
+                const forward = new Vector3(0, 0, 1);
+                const direction = this._rotationCircle.getDirection(forward);
+                direction.normalize();
+
+                // Remove Rotation Circle from parent mesh before drag interaction
+                this._gizmoMesh.removeChild(this._rotationCircle);
+
+                lastDragPosition.copyFrom(e.dragPlanePoint);
+                dragPlanePoint = e.dragPlanePoint;
+                const origin = this._rotationCircle.getAbsolutePosition().clone();
+                const originalRotationVector = this._rotationCircle.getAbsolutePosition().clone().addInPlace(direction);
+                const dragStartVector = e.dragPlanePoint;
+                let angle = this.angleBetween3DCoords(origin, originalRotationVector, dragStartVector);
```

Comment:
this can be replaced with a call to `GetAngleBetweenVectors` with (originalRotationVector.subtract(origin), dragStartVector.subtract(origin), this._rotationCircle.up). also, originalRotationVector and dragStartVector are misleading. they are points, not vectors.